### ISSUE TITLE
Kodi 19.4 choppy video and audio out of sync since Chromecast upgrade to Android 12

### ISSUE THREAD

**Issue by timkgh** [xbmc/xbmc#22072]
### Describe the bug
Here is a clear and concise description of what the problem is:

Since the upgrade to Android 12, Kodi 19.4 on Google Chromecast 4K has choppy video and audio out of sync when pass-through is enabled. I tried frame matching enabled and disabled both in Chromecast and Kodi and it does not help. The only way to make it work is to disable audio pass-through. This happens on 1080p or 4K content with DD 5.1 audio, no HDR or DV involved.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Playing should be smooth, audio should be in sync, in particular now that Chromecast 4K _claims_ to support frame rate matching.

## Possible Fix

Workaround: disable audio pass-through

### To Reproduce

1. 
2.
3.

### Debuglog

The debuglog can be found here:

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [x] Android 12 on Google Chromecast 4K
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] OSX
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version: 19.4

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by CastagnaIT** [comment]
Please provide full log of a played stream

**Comment by gerarg** [comment]
I am facing similar issues with my Chromecast 4 since the Android 12 update, with passthrough enabled files with DTS or DTS-HD don't output any audio, DD+ and DD 5.1 are very choppy and TrueHD is playing as PCM on my receiver (I remember it played as DD+ before).

If I select Audiotrack(RAW) instead of Audiotrack(IEC) then DD, DD+ and DTS files work fine but TrueHD is still playing as PCM and DTS-HD plays as DTS.

Tried changing about every setting I could find and even doing a factory reset but nothing worked.

**Comment by fritsch** [comment]
Choose: Audiotrack (RAW)
Enable: DTS, DD, DD+
Further Enable: Dolby Transcoding
Set Speaker Channels to 2.0 (this is important)

Then you should get:
DD+, DD, DTS-Core from DTS-HD and DD from TrueHD.

Btw. kodi has a lossless dTS-HD and TrueHD decoder - whenever you device supports multi pcm output ... there is no need to transcode it to DD.

For the choppiness of IEC, you need to open another defect. Seems that the device does not properly report Delays with the standard API from what you tell.

**Comment by timkgh** [comment]
@fritsch I never understood the reasoning behind these settings. Do they apply to all Android devices or just Chromecast? Is there a good guide that explains it well?

> Further Enable: Dolby Transcoding

This is not transcoding DTS to DD, correct? Otherwise what is it transcoding, something like multi-channel AAC? But then why not decode it and send it as multi PCM over HDMI?

> kodi has a lossless dTS-HD and TrueHD decoder - **whenever you device supports multi pcm output** ... there is no need to transcode it to DD.

What should the speaker channels be set to to get multi PCM over HDMI?

**Comment by fritsch** [comment]
For last question: 5.1 or 7.1
then Dolby transcoding setting won't be available anymore. As you rightly said, it makes not much sense to reencode it if you can output it directly.

Now comes the troublesome truth:
Many android boxes.cannot do lpcm at all, they will only out stereo :-( for this and old spdif devices we provide that workaround: 2 pcm channels only but AC3 5.1 still having at least 6 channels when device cannot PT Truehd ...

As you get pcm stereo for Truehd your device is most likely one of those.

**Comment by timkgh** [comment]
> Many android boxes.cannot do lpcm at all, they will only out stereo :-(

Yikes. I understand now why sometimes 2.0 is the recommended setting anyway. IIRC, Fire TV stick 4K broke things at some point, not sure whether it's still broken or they fixed it. I suppose I need to always have test media files on hand and try it.

**Comment by PiotrDx** [comment]
I have very similar issue.
Tested on Kodi 19.3-19.5 and 20.x.
My player is Nokia Streaming Box (Android TV 10) => Onkyo AV => TV.

If select Audiotrack(IEC) then DD, DD+ and DTS are choppy (video and audio is desynchronized)
If select  Audiotrack(RAW) then DD, DD+ is ok, but DTS is not played (AV does not recognize it).

When using old Xiaomi Mi Box (Android TV 9) everything is ok.

**Comment by markdj57** [comment]
Sounds like same issue. I cannot get DTS output to work, but I'm running Chromecast 4k (Google TV) with kodi 15.3. I can change audio stream from a specific video file to ac-3 and it will show DD5.1 on the amp, but DTS is never output. I've also noticed the sound delay sync to be anywhere from 1.5 to 2 seconds, never needed it before. I hope this is an android thing that will be resolved. I've tried VLC and same thing, DTS won't pass through.
